tests/test_core.py

```diff
@@ -8370,10 +8373,9 @@ def setUp(self):
 strict = make_run('strict', emcc_args=[], settings={'STRICT': 1})
 
 lsan = make_run('lsan', emcc_args=['-fsanitize=leak', '-O2'], settings={'ALLOW_MEMORY_GROWTH': 1})
-# default MAXIMUM_MEMORY to 7/8 of 2GB to allow plenty of room for shadow memory
-asan = make_run('asan', emcc_args=['-fsanitize=address', '-O2'], settings={'ALLOW_MEMORY_GROWTH': 1, 'MAXIMUM_MEMORY': 1879048192})
+asan = make_run('asan', emcc_args=['-fsanitize=address', '-O2'], settings={'ALLOW_MEMORY_GROWTH': 1, 'INITIAL_MEMORY': 314572800})
 asani = make_run('asani', emcc_args=['-fsanitize=address', '-O2', '--pre-js', os.path.join(os.path.dirname(__file__), 'asan-no-leak.js')],
-                 settings={'ALLOW_MEMORY_GROWTH': 1, 'MAXIMUM_MEMORY': 1879048192})
+                 settings={'ALLOW_MEMORY_GROWTH': 1, 'INITIAL_MEMORY': 314572800})
```

Comment:
does this mean that without setting MAXIMUM_MEMORY we use the default 2GB, which means shadow memory is 256MB?

if so that may not be best for CI, as each test will use that much memory immediately. maybe worth keeping the maximum?

Reply:
Yes, this goes back to using the default MAXIMUM_MEMORY, but we were already doing that before this PR. The MAXIMUM_MEMORY that we used to have in this PR was just 7/8 of the default, so not a whole lot better.

I could believe that this might cause CI issues, but it hasn't so far, so I think that would be better addressed in a separate PR.

Reply:
Oh, I was reading the diff wrong. sgtm.

Last question: why add `-O2` here?

Reply:
The SIMD tests were failing because `-O0` combined with ASan was leading to functions with too many locals. Those failures aren't related to this change, so I could remove the `-O2` from this PR, but adding it didn't seem to have any downsides, either.

Reply:
I see, sounds fine.